internal/backend/crypto/gpg/cli/recipients.go

```diff
@@ -45,6 +46,28 @@ func (g *GPG) FindRecipients(ctx context.Context, search ...string) ([]string, e
 	if gpg.IsAlwaysTrust(ctx) {
 		recp = kl.Recipients()
 	}
+	debug.Log("recp before subkey check: %q", recp)
+
+	// let us support the ! syntax that enforces specific subkey usage
```

Comment:
This is bit too deeply nested for my taste.

I believe we can limit nesting by inverting that match conditions and using `continue` for the non-match cases.

Something like this (pseudo):

```
for _, val := range search {
  str := ...
  if == "" || !strings.HasSuffix(val, "!") {
    continue
  }
  for _, key := range kl {
    for sub := range key.SubKeys {
      if !strings.Contains(...) {
        continue
      }
      recp = slices ...
```